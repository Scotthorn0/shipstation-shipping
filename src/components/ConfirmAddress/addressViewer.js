import React from 'react';
import PropTypes from 'prop-types';
import { List } from 'semantic-ui-react';

const AddressViewer = ({ address }) => (
  <List>
    {
      Object.keys(address).map(key => (
        address[key] && key !== 'address_residential_indicator' ? <List.Item key={key}>{address[key]}</List.Item> : null))
    }
  </List>
);

AddressViewer.propTypes = {
  address: PropTypes.object,
};

export default AddressViewer;
