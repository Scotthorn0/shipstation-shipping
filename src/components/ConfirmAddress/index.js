import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Header, Segment, Button } from 'semantic-ui-react';
import AddressViewer from './addressViewer';
import { NEXT, BACK } from '../../constants';
import './styles.css';

class ConfirmAddress extends Component {
  nextStep = () => {
    const { moveStep } = this.props;
    moveStep(NEXT);
  }

  previousStep = () => {
    const { moveStep } = this.props;
    moveStep(BACK);
  }

  render() {
    const { requestedAddress, verifiedAddress } = this.props;
    return (
      <Segment basic>
        <Header content="Confirm the Address is correct." />
        <Segment.Group horizontal>
          <Segment>
            <Header size="small" content="What you entered" />
            <AddressViewer address={requestedAddress} />
          </Segment>
          <Segment>
            <Header size="small" content="What we found" />
            <AddressViewer address={verifiedAddress} />
          </Segment>
        </Segment.Group>
        <Segment basic textAlign="right">
          <Header textAlign="center" size="medium" content="Is this correct?" />
          <Button
            icon="left arrow"
            negative
            labelPosition="left"
            content="TryAgain"
            onClick={this.previousStep}
          />
          <Button
            icon="check"
            positive
            labelPosition="right"
            content="Yes"
            onClick={this.nextStep}
          />
        </Segment>
      </Segment>
    );
  }
}

ConfirmAddress.propTypes = {
  requestedAddress: PropTypes.object,
  verifiedAddress: PropTypes.object,
  moveStep: PropTypes.func,
};

export default ConfirmAddress;
