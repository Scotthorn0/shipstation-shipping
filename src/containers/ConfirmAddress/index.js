import { connect } from 'react-redux';
import ConfirmComponent from '../../components/ConfirmAddress';

export const mapStateToProps = ({ requestedAddress, verifiedAddress }) => ({
  requestedAddress,
  verifiedAddress,
});

export default connect(mapStateToProps)(ConfirmComponent);
