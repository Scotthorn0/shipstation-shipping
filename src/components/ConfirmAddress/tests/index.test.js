import React from 'react';
import { shallow } from 'enzyme';
import ConfirmAddress from '../';
import { NEXT, BACK } from '../../../constants';

describe('ConfirmAddress Component', () => {
  let wrapper;
  let instance;
  let moveStepMock;

  beforeEach(() => {
    moveStepMock = jest.fn();
    wrapper = shallow(<ConfirmAddress moveStep={moveStepMock} />);
    instance = wrapper.instance();
  });

  test('should render without crashing', () => {
    expect(wrapper.exists()).toBe.true;
  });

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('nextStep', () => {
    test(`should call moveStep with ${NEXT}`, () => {
      instance.nextStep();
      expect(moveStepMock.mock.calls[0][0]).toBe(NEXT);
    });
  });

  describe('previousStep', () => {
    test(`should call moveStep with ${BACK}`, () => {
      instance.previousStep();
      expect(moveStepMock.mock.calls[0][0]).toBe(BACK);
    });
  });
});
