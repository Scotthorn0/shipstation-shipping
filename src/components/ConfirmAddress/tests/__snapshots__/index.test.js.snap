// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmAddress Component should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmAddress
    moveStep={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "basic": true,
      "children": Array [
        <Header
          content="Confirm the Address is correct."
/>,
        <SegmentGroup
          horizontal={true}
>
          <Segment>
                    <Header
                              content="What you entered"
                              size="small"
                    />
                    <AddressViewer
                              address={undefined}
                    />
          </Segment>
          <Segment>
                    <Header
                              content="What we found"
                              size="small"
                    />
                    <AddressViewer
                              address={undefined}
                    />
          </Segment>
</SegmentGroup>,
        <Segment
          basic={true}
          textAlign="right"
>
          <Header
                    content="Is this correct?"
                    size="medium"
                    textAlign="center"
          />
          <Button
                    as="button"
                    content="TryAgain"
                    icon="left arrow"
                    labelPosition="left"
                    negative={true}
                    onClick={[Function]}
          />
          <Button
                    as="button"
                    content="Yes"
                    icon="check"
                    labelPosition="right"
                    onClick={[Function]}
                    positive={true}
          />
</Segment>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": "Confirm the Address is correct.",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Segment>
              <Header
                            content="What you entered"
                            size="small"
              />
              <AddressViewer
                            address={undefined}
              />
</Segment>,
            <Segment>
              <Header
                            content="What we found"
                            size="small"
              />
              <AddressViewer
                            address={undefined}
              />
</Segment>,
          ],
          "horizontal": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Header
                  content="What you entered"
                  size="small"
/>,
                <AddressViewer
                  address={undefined}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "content": "What you entered",
                  "size": "small",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "address": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Header
                  content="What we found"
                  size="small"
/>,
                <AddressViewer
                  address={undefined}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "content": "What we found",
                  "size": "small",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "address": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "basic": true,
          "children": Array [
            <Header
              content="Is this correct?"
              size="medium"
              textAlign="center"
/>,
            <Button
              as="button"
              content="TryAgain"
              icon="left arrow"
              labelPosition="left"
              negative={true}
              onClick={[Function]}
/>,
            <Button
              as="button"
              content="Yes"
              icon="check"
              labelPosition="right"
              onClick={[Function]}
              positive={true}
/>,
          ],
          "textAlign": "right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "content": "Is this correct?",
              "size": "medium",
              "textAlign": "center",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "content": "TryAgain",
              "icon": "left arrow",
              "labelPosition": "left",
              "negative": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "content": "Yes",
              "icon": "check",
              "labelPosition": "right",
              "onClick": [Function],
              "positive": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "basic": true,
        "children": Array [
          <Header
            content="Confirm the Address is correct."
/>,
          <SegmentGroup
            horizontal={true}
>
            <Segment>
                        <Header
                                    content="What you entered"
                                    size="small"
                        />
                        <AddressViewer
                                    address={undefined}
                        />
            </Segment>
            <Segment>
                        <Header
                                    content="What we found"
                                    size="small"
                        />
                        <AddressViewer
                                    address={undefined}
                        />
            </Segment>
</SegmentGroup>,
          <Segment
            basic={true}
            textAlign="right"
>
            <Header
                        content="Is this correct?"
                        size="medium"
                        textAlign="center"
            />
            <Button
                        as="button"
                        content="TryAgain"
                        icon="left arrow"
                        labelPosition="left"
                        negative={true}
                        onClick={[Function]}
            />
            <Button
                        as="button"
                        content="Yes"
                        icon="check"
                        labelPosition="right"
                        onClick={[Function]}
                        positive={true}
            />
</Segment>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": "Confirm the Address is correct.",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Segment>
                <Header
                                content="What you entered"
                                size="small"
                />
                <AddressViewer
                                address={undefined}
                />
</Segment>,
              <Segment>
                <Header
                                content="What we found"
                                size="small"
                />
                <AddressViewer
                                address={undefined}
                />
</Segment>,
            ],
            "horizontal": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Header
                    content="What you entered"
                    size="small"
/>,
                  <AddressViewer
                    address={undefined}
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "content": "What you entered",
                    "size": "small",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "address": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Header
                    content="What we found"
                    size="small"
/>,
                  <AddressViewer
                    address={undefined}
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "content": "What we found",
                    "size": "small",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "address": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "basic": true,
            "children": Array [
              <Header
                content="Is this correct?"
                size="medium"
                textAlign="center"
/>,
              <Button
                as="button"
                content="TryAgain"
                icon="left arrow"
                labelPosition="left"
                negative={true}
                onClick={[Function]}
/>,
              <Button
                as="button"
                content="Yes"
                icon="check"
                labelPosition="right"
                onClick={[Function]}
                positive={true}
/>,
            ],
            "textAlign": "right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "content": "Is this correct?",
                "size": "medium",
                "textAlign": "center",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "content": "TryAgain",
                "icon": "left arrow",
                "labelPosition": "left",
                "negative": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "content": "Yes",
                "icon": "check",
                "labelPosition": "right",
                "onClick": [Function],
                "positive": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
